civicrm_views_integration/civicrm_views_integration.module

```diff
@@ -1,27 +1,58 @@
 <?php
 
 /**
- * Implements CiviCRM hook :  hook_civicrm_post() [https://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_post].
- * This hook will check if any custom group get added and will append the prefix for it to settings.php
+ * Implements CiviCRM hook : hook_civicrm_post() [https://wiki.civicrm.org/confluence/display/CRMDOC/hook_civicrm_post].
+ * This hook will check if any custom group get added and will create database prefix for it to be used in settings.php
  * file automatically
  */
 function civicrm_views_integration_post($op, $objectName, $id, &$params) {
   if ($objectName == 'CustomGroup' && $op == 'create') {
-    $config = CRM_Core_Config::singleton();
+    _civicrm_views_integrating_generate_db_prefixes();
+  }
+}
+
+/**
+ * Creates Views integration settings database prefix array to be used in settings.php 
+ * for connecting CiviCRM to views.
+ */
+function _civicrm_views_integrating_generate_db_prefixes(){
+
+  civicrm_initialize();
+  global $databases;
 
-    $tableName = CRM_Core_DAO::getFieldValue( 'CRM_Core_DAO_CustomGroup', $id, 'table_name' );
+  $config = CRM_Core_Config::singleton();
 
-    // Only add the prefix if civicrm and drupal database are not the same one
-    if ($config->dsn != $config->userFrameworkDSN) {
-      $dsnArray = DB::parseDSN($config->dsn);
-      $prefix = "`{$dsnArray['database']}`.";
+  // Only add the prefix if civicrm and drupal database are not the same one
+  if ($config->dsn != $config->userFrameworkDSN) {
+    $tableNames = CRM_Core_DAO::GetStorageValues(NULL, 0, 'Name');
 
-      $ctable = '$databases[\'default\'][\'default\'][\'prefix\'][\'' . $tableName . '\']= ' . '\'' . $prefix . '\';';
+    $tablePrefixes = array();
+    // add default prefix: the drupal database prefix
+    $tablePrefixes['default'] = '';
 
-      $cmsSettingsPath = $config->userSystem->cmsRootPath() . '/sites/default/settings.php';
-      $myfile = fopen($cmsSettingsPath, 'a');
-      fwrite($myfile, "\n". $ctable);
-      fclose($myfile);
+    $dsnArray = DB::parseDSN($config->dsn);
+    $prefix = "`{$dsnArray['database']}`.";
+    
+    foreach ($tableNames as $tableName => $value) {
+      $tablePrefixes[$tableName] = "'" . $prefix . "',";;
     }
+    // Generates JSON file.
+    _civicrm_views_integrating_generate_json($tablePrefixes, $config);
   }
-}
\ No newline at end of file
+}
+
+/**
+ * Creates Json file to hold Views integration settings database prefix array to be used in settings.php 
+ * @param array
+ *      Database prefix array.
+ * @param object
+ *      CiviCRM config object.
+ */
+function _civicrm_views_integrating_generate_json($tablePrefixes = array(), $config){
+
+  $json_data = json_encode($tablePrefixes);
+  $cmsSettingsPath = $config->userSystem->cmsRootPath() . '/sites/default/files/civicrm/custom/views-mapping.json';
```

Comment:
you have used (DRUPAL_ROOT . '/sites/default/files/civicrm/custom/views-mapping.json') ... why not using the same thing here ... and then you don't even need to pass $config to this function